starter-core/src/main/java/io/micronaut/starter/feature/function/gcp/GoogleCloudEventsFunction.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.function.gcp;
+
+import com.fizzed.rocker.RockerModel;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.Project;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.feature.FeatureContext;
+import io.micronaut.starter.feature.function.gcp.template.gcpFunctionReadme;
+import io.micronaut.starter.feature.function.gcp.template.cloudevents.*;
+import io.micronaut.starter.feature.other.ShadePlugin;
+import io.micronaut.starter.options.BuildTool;
+import io.micronaut.starter.options.Language;
+import io.micronaut.starter.template.RockerTemplate;
+import jakarta.inject.Singleton;
+import java.util.Optional;
+
+
+/**
+ * A feature for supporting Google CloudEvents Function.
+ *
+ * @author Guillermo Calvo
+ * @since XXX
+ */
+@Singleton
+public class GoogleCloudEventsFunction extends AbstractGoogleCloudFunction {
```

Comment:
overrides `supports(ApplicationType)` method and return false unless `ApplicationType.FUNCTION`